packages/core/src/dialects/abstract/query-interface.types.ts

```diff
@@ -1,21 +1,36 @@
+import type { Literal } from '../../expression-builders/literal';
 import type { QueryRawOptions, Sequelize } from '../../sequelize';
 import type { AbstractQueryGenerator, CreateSchemaQueryOptions, ListSchemasQueryOptions } from './query-generator';
+import type { TruncateTableOptions } from './query-generator.types';
+
+interface Replaceable {
+  /**
+   * Only named replacements are allowed in query interface methods.
+   */
+  replacements?: { [key: string]: unknown };
+}
 
 export interface QueryInterfaceOptions {
   sequelize: Sequelize;
   queryGenerator: AbstractQueryGenerator;
 }
 
-/** Options accepted by {@link AbstractQueryInterfaceTypeScript#createSchema} */
+/** Options accepted by {@link AbstractQueryInterface#createSchema} */
 export interface CreateSchemaOptions extends CreateSchemaQueryOptions, QueryRawOptions { }
 
-/** Options accepted by {@link AbstractQueryInterfaceTypeScript#showAllSchemas} */
+/** Options accepted by {@link AbstractQueryInterface#showAllSchemas} */
 export interface ShowAllSchemasOptions extends ListSchemasQueryOptions, QueryRawOptions { }
 
-/** Options accepted by {@link AbstractQueryInterfaceTypeScript#dropAllSchemas} */
+/** Options accepted by {@link AbstractQueryInterface#dropAllSchemas} */
 export interface DropAllSchemasOptions extends QueryRawOptions {
   /**
    * List of schemas to skip dropping (i.e., list of schemas to keep)
    */
   skip?: string[];
 }
+
+/** Options accepted by {@link AbstractQueryInterface#bulkDelete} */
+export interface BulkDeleteOptions extends QueryRawOptions, Replaceable, TruncateTableOptions {
+  limit?: number | Literal | null | undefined;
```

Comment:
Do we need to set both `null` and `undefined` when it's already optional?

Reply:
No, I've updated this now.